---
title: 'R fridays #8: PRO-seq and heatmaps'
author: Sarah Arcos
date: '2019-08-23'
slug: r-fridays-8-pro-seq-and-heatmaps
categories: []
tags:
  - PRO-ses
  - RNA-seq
excerpt: "Monica demonstrates how to make heatmaps from PRO-seq data using the pheatmap package."
---



<div id="looking-for-patterns-in-pro-seq-with-heatmaps" class="section level1">
<h1>Looking for patterns in PRO-seq with heatmaps</h1>
<p>Traditional RNA-seq measures RNA <strong>levels</strong> which are the result of both transcription and degradation of mRNA transcripts. <a href="https://lab.vanderbilt.edu/hiebert-lab/person/monica-bomber/">Monica Bomber</a> and <a href="https://lab.vanderbilt.edu/hiebert-lab/person/hillary-layden/">Hillary Layden</a> of the Hiebert Lab use <a href="https://www.nature.com/articles/nprot.2016.086">PRO-seq</a> (precision nuclear run-on sequencing) to directly measure gene transcription rates. PRO-seq can measure transcription rates and identify transcription start sites at high-resolution througout the genome. Today, Monica and Hillary showed us how to generate a heatmap to understand how transcription rates change over time in a human B cell lymphoma cell line. Take a look at the approach and simulated data below.</p>
<div id="the-experiment" class="section level2">
<h2>The experiment</h2>
<p>The Hiebert lab is interested in changes in transcription after depletion of a protein called FOXO1 in human B cell lymphoma cells (DHL-4). Hillary and Monica used the <a href="https://www.nature.com/articles/s41589-018-0021-8">dTAG</a> system to degrade FOXO1, and then measured changes in transcription over time. They looked at 5 timepoints after degradation of FOXO1 (0.5 hours, 1 hour, 2 hours, 6 hours, and 24 hours).</p>
</div>
<div id="reading-in-the-data" class="section level2">
<h2>Reading in the data</h2>
<p>The data was read in using <code>read_csv()</code> from the tidyverse.</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre class="r"><code>proseq &lt;- read_csv(&quot;../../data/proseq_082319_noise.csv&quot;)</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   `Transcript ID (duplicates removed)` = col_double(),
##   `0.5 hrs` = col_double(),
##   `1 hr` = col_double(),
##   `2 hrs` = col_double(),
##   `6 hrs` = col_double(),
##   `24 hrs` = col_double()
## )</code></pre>
<p>This table contains the log2-transformed expression values for the untreated cells and for each timepoint after degradation of FOXO1.</p>
</div>
<div id="making-a-heatmap-with-pheatmap" class="section level2">
<h2>Making a heatmap with <a href="https://cran.r-project.org/web/packages/pheatmap/pheatmap.pdf">pheatmap</a></h2>
<p>Monica and Hillary wanted to look for patterns in trancriptional changes over the course of their experiment. To do this they used the <a href="https://cran.r-project.org/web/packages/pheatmap/pheatmap.pdf">pheatmap</a> package to draw a heatmap and the <a href="https://cran.r-project.org/web/packages/RColorBrewer/RColorBrewer.pdf">RColorBrewer</a> package to pick aesthetically pleasing colors.</p>
<pre class="r"><code>library(pheatmap)
library(RColorBrewer)</code></pre>
<p>First, they removed the first column of their data, which contains gene names.</p>
<pre class="r"><code>proseq &lt;- select(proseq, -contains(&quot;Transcript&quot;))</code></pre>
<p>Next they specified the color gradient to use for the heatmap</p>
<pre class="r"><code>my_palette=colorRampPalette(rev(brewer.pal(n = 11, name = &quot;RdYlBu&quot;)))(1000)</code></pre>
<p>Finally they generated a heatmap using the <code>pheatmap()</code> function from <a href="https://cran.r-project.org/web/packages/pheatmap/pheatmap.pdf">pheatmap</a>.</p>
<pre class="r"><code>pheatmap(proseq, 
         color = my_palette,
         cellwidth = 60, 
         cellheight = 2, 
         cluster_cols=F, 
         cutree_rows=7, 
         scale=&quot;row&quot;,
         fontsize_row=2)</code></pre>
<p><img src="/post/2019-09-23-r-fridays-8-pro-seq-and-heatmaps_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>By default the <code>pheatmap()</code> function performs <a href="https://stat.ethz.ch/R-manual/R-devel/library/stats/html/hclust.html">hierarchical clustering</a> on the input data before drawing the heatmap. From the clustering and the visualization, there are clear patterns of genes that have increased or decreased transcriptional activity over time after depletion of FOXO1.</p>
<p>The <code>pheatmap()</code> function has many options to customize heatmaps, like adding annotations to the cells and using different clustering algorithms (like k-means). Dave Tang recently wrote <a href="https://davetang.org/muse/2018/05/15/making-a-heatmap-in-r-with-the-pheatmap-package/">an excellent blog post</a> going into more detail about using <code>pheatmap()</code> to examine RNA-seq data.</p>
</div>
</div>
<div id="wrap-up" class="section level1">
<h1>Wrap-up</h1>
<p>Today was a short introduction to drawing heatmaps to look at time-series data. In the future we will go into more detail about different kinds of clustering and when it is appropriate to use them.</p>
<p><em>As always, if you have any questions donâ€™t hesitate to <a href="https://saraharcos.github.io/contact.html">email me</a>.</em></p>
</div>
